src/devices/chihiro/JvsIo.cpp

```diff
@@ -97,203 +118,246 @@ int JvsIo::Jvs_Command_GetBoardId()
 			uint8_t value
-				= (i >= 2) ? 0 // We only have two coin slots, if a title should ask for more, pad with dummy data
-				: (j == 0) ? Inputs.coins[i].GetByte0()
-				: (j == 1) ? Inputs.coins[i].GetByte1()
-				: 0;
+				= (j == 0) ? coin_slot.GetByte0()
+				: (j == 1) ? coin_slot.GetByte1()
+				: 0; // Pad any remaining bytes with 0, as we don't have that many inputs available
 			ResponseBuffer.push_back(value);
 		}
 	}
 
 	return 1;
 }
 
-int JvsIo::Jvs_Command_ReadAnalogInputs(uint8_t* data)
+int JvsIo::Jvs_Command_22_ReadAnalogInputs(uint8_t* data)
 {
-	uint8_t inputs = data[1];
+	static jvs_analog_input_t default_analog;
+	uint8_t nr_analog_inputs = data[1];
 
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 
-	for (int i = 0; i < inputs; i++) {
-		const uint8_t bytesPerInput = 2;
-		for (int j = 0; j < bytesPerInput; j++) {
+	for (int i = 0; i < nr_analog_inputs; i++) {
+		const uint8_t bytesPerAnalogInput = 2;
+		for (int j = 0; j < bytesPerAnalogInput; j++) {
+			// If a title asks for more analog input than we support, pad with dummy data
+			jvs_analog_input_t &analog_input = (i >= JVS_MAX_ANALOG) ? default_analog : Inputs.analog[i];
 			uint8_t value
-				= (i >= 8) ? ((j == 0) ? 0x80 : 0x00) // We only have 8 analog inputs, if a title should ask for more, pad with dummy data
-				: (j == 0) ? Inputs.analog[i].GetByte0()
-				: (j == 1) ? Inputs.analog[i].GetByte1()
-				: 0;
+				= (j == 0) ? analog_input.GetByte0()
+				: (j == 1) ? analog_input.GetByte1()
+				: 0; // Pad any remaining bytes with 0, as we don't have that many inputs available
 			ResponseBuffer.push_back(value);
 		}
 	}
 
 	return 1;
 }
 
-int JvsIo::Jvs_Command_GeneralPurposeOutput(uint8_t* data)
+int JvsIo::Jvs_Command_32_GeneralPurposeOutput(uint8_t* data)
 {
 	uint8_t banks = data[1];
 
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 
 	// TODO: Handle output
 
-	// Data size is n.banks followed by one byte per bank
+	// Data size is 1 byte indicating the number of banks, followed by one byte per bank
 	return 1 + banks;
 }
 
-void JvsIo::HandlePacket(jvs_packet_header_t* header, uint8_t* payload)
+uint8_t JvsIo::GetEscapedByte(uint8_t* &payload)
 {
-	// Decode the payload data
-	std::vector<uint8_t> packet;
-	for (unsigned i = 0; i < header->count; i++) {
-		uint8_t value = payload[i];
+	uint8_t value = *payload++;
 
-		// Special case: 0xD0 is an exception byte that actually returns the next byte + 1
-		if (value == 0xD0) {
-			value = payload[i++] + 1;
-		}
-
-		packet.push_back(value);
+	// Special case: 0xD0 is an exception byte that actually returns the next byte + 1
+	if (value = EscapeByte) {
+		value = *payload++ + 1;
 	}
 
+	return value;
+}
+
+void JvsIo::HandlePacket(jvs_packet_header_t* header, std::vector<uint8_t>& packet)
+{
 	// Clear the response buffer
 	ResponseBuffer.clear();
 
 	ResponseBuffer.push_back(0x01);

```

Comment:
You defined the StatusCode enum but didn't use it here?

Reply:
Hm, something to expand upon indeed